tests/v2/validation/snapshot/snapshot.go

```diff
@@ -59,16 +67,14 @@ func snapshotRestore(t *testing.T, client *rancher.Client, clusterName string, e
 	localClusterID, err := clusters.GetClusterIDByName(client, localClusterName)
 	require.NoError(t, err)
 
-	var isRKE1 bool
+	var isRKE1 = false
 
 	clusterObject, _, _ := clusters.GetProvisioningClusterByName(client, clusterName, namespace)
```

Comment:
Maybe I do not understand this, but even for RKE2, and K3s there are v3 objects right? Does it distinguish it via name vs ID?

We have this extension and I think it can work pretty well here: https://github.com/rancher/shepherd/blob/main/extensions/clusters/clustermeta.go#L52-L76

This is gonna return provider type and other options: https://github.com/rancher/shepherd/blob/main/extensions/clusters/clustermeta.go#L31-L49

So that tests don't need to assume if there's successful v3 it's an RKE1. 

What do you all think?

Reply:
Yeah, it distinguishes by the ID, so it ensures that the actual cluster ID will be rke1 or v2 prov cluster.